File: 08challenge/main.c
#include <stdio.h>

int main() {
    float distance;
    printf("Enter a distance in meters: ");
    scanf("%f", &distance);
    float km = distance / 1000;
    float dam = distance / 10;
    float hm = distance / 100;
    float dm = distance * 10;
    float cm = distance * 100;
    float mm = distance * 1000;
    printf("The distance of %.1fm corresponds to:\n", distance);
    printf("%.5fKm\n", km);
    printf("%.4fDam\n", dam);
    printf("%.4fHm\n", hm);
    printf("%.1fDm\n", dm);
    printf("%.0fCm\n", cm);
    printf("%.0fMm\n", mm);
    return 0;
}

File: 08challenge/main.go
package main

import (
    "fmt"
    "strconv"
)

func main() {
    var distance float64
    fmt.Print("Enter a distance in meters: ")
    fmt.Scan(&distance)

    km := distance / 1000
    dam := distance / 10
    hm := distance / 100
    dm := distance * 10
    cm := distance * 100
    mm := distance * 1000

    fmt.Println("The distance of " + strconv.FormatFloat(distance, 'f', 1, 64) + "m corresponds to:")
    fmt.Printf("%.5fKm\n", km)
    fmt.Printf("%.4fDam\n", dam)
    fmt.Printf("%.4fHm\n", hm)
    fmt.Printf("%.1fDm\n", dm)
    fmt.Printf("%.0fCm\n", cm)
    fmt.Printf("%.0fMm\n", mm)
}
